### PR TITLE
Fixed getting Folders by Permissions

### LINKED ISSUE #4573: Resource Manager permissions bug/issue in DNN 9.8.1 and above (HTML Editor Browse Server showing all folders)
I believe the Resource Manager has a folder permissions bug/issue. The HTML Editor "Browse Server" is showing ALL folders to Roles after upgrading from DNN 9.7.2 to 9.8.1 no matter how the permissions are set in the Resource Manager.

When using the HTML Editor, selecting the image or file icon, selecting the "Browse Server" button. The Folder Tree shows all folders no matter what the folder permissions are set to for Roles in the Resource Manager. In DNN 9.7.2 I had folder permissions set so that I could hide some of the folders from certain Roles, but after upgrading to 9.8.1 it shows all the folders.

I have tried different folder permission settings in the PB > Site Assets, but nothing works. I have even tried replacing with the new Resource Manger but that did not work either.

I have tested this on a test production site and a test dnndev.me site and both have the same issue.

**To replicate the issue. Go to the Site Assets and assign folder permissions on a folder to hide it from a "Role". Then login as a user assigned to that Role and use the HTML module "Browse Server" and you will see the folder you are trying to hide.**

I have attached a couple of screen shots.

DNN 9 8 1 Site Assets Folder permissions

DNN 9 8 1 HTML Editor Browse Server showing all folders

Thanks,
Craig

### PR DESCRIPTION
**Pull request by yog-it**
Fixes #4573

## Summary
Retrieving folders per permissions needed to include conditionals for both grant and deny permissions; also using CTE instead of temp table for better performance.

### CODE REVIEW

File: DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.10.03.SqlDataProvider [hunk 1]
```diff
@@ -0,0 +1,125 @@
+﻿/****************************************************************
+ * SPROC: GetFoldersByPermission #4573
```
Comment (david-poindexter) [inline DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.10.03.SqlDataProvider:2]:
```suggestion
 * SPROC: GetFoldersByPermission
```

**Review by david-poindexter — CHANGES_REQUESTED** [review]
I didn't have time to test out this PR, but I did notice the GitHub issue number here and think we should remove probably remove it.  We'll have a record of it in the git history.

**Review by valadas — APPROVED** [review]
It's really funny because I was just looking into this today for a couple of hours and then came here and noticed this PR.

I did run some tests locally and it looks like it's behaving the way it should. Great work!

File: DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.10.03.SqlDataProvider (hunk 1 above)
Comment (valadas) [reply DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.10.03.SqlDataProvider:2]: I agree

**Review by mitchelsellers — APPROVED** [review]
I too tested this one, and it seems to be working well

**Review by david-poindexter — APPROVED** [review]

**Comment by cgentryls** [comment]
@yog-it thank you for working on and resolving this! I really appreciate it! And, I would like to thank all of you for your continued work on DNN! I inherited our DNN site a little over 6 years ago upgrading it from 4.9.5 to 7.4.1. Currently we are at 9.7.2. I work with other CMS platforms and DNN is by far the best! Thanks again and long live DNN!

**Comment by yog-it** [comment]
@cgentryls my pleasure.  I too have been a long time DNN user and agree completely, it's just one of the best open source CMS platforms out there. I started contributing only a short while ago, but love working with the project and helping out other DNN'ers.

**Comment by david-poindexter** [comment]
@cgentryls and @yog-it thanks so much for your comments - this is what fuels us all!  DNN Platform is fantastic and the community is "fantasticer"!!!  😁🎉

**Pull request merged**